community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/input/csv/ExternalPropertiesDecorator.java

```diff
@@ -50,11 +48,11 @@
  * Then properties {@code abc@somewhere} and {@code def@somewhere} will be amended to input node {@code 1}
  * and {@code ghi@someplace} to input node {@code 3}.
  */
-public class ExternalPropertiesDecorator implements Function<InputNode,InputNode>
+public class ExternalPropertiesDecorator implements Decorator<InputNode>
```

Comment:
`ExternalPropertiesDecorator` assumes input order is the same (in terms of ids) as the new input data (the properties). This should at least be documented with a recommendation to use sorted data.